CMSgov/easi-app
# [NOREF] Add 'submitted_at' field to TRB request forms

# NOREF

## Changes and Description

- Added a `submitted_at` field to TRB Request forms. This value will be set to the current time whenever marking a request as "completed" (submitting multiple times will only store the last submission date)

## How to test this change

Create a TRB request
```graphql
mutation {
  createTRBRequest(requestType: NEED_HELP) {
    id # use this for the following steps
    form {
      status
    }
    taskStatuses {
      formStatus
      feedbackStatus
      consultPrepStatus
    }
  }
}
```

Note that there's no submission time on the form
```graphql
query get {
  trbRequest(id: "idfrombefore") {
    form {
      status
      submittedAt
    }
  }
}
```

See there's a submission time!
```graphql
mutation submit {
  updateTRBRequestForm(input: {
    trbRequestId: "idfrombefore"
    isSubmitted: true
  }) {
    trbRequestId
    status
    submittedAt
  }
}
```

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
